### PR TITLE
Idempotent, non-interactive launch

### PR DESCRIPTION
**Pull request by giorgiosironi**
```
[11:22:20][giorgio@Newton:~/code/builder]$ bldr launch:journal,pr-test
...
INFO - cfn - stack journal--pr-test already exists

Done.
```

### CODE REVIEW

**Comment by giorgiosironi** [comment]
Most of the PR is making the tests run fine with the recent parallelization.

File: src/buildercore/checks.py [hunk 1]
```diff
@@ -9,7 +9,9 @@ class AccessProblem(RuntimeError):
     pass
 
 class StackAlreadyExistsProblem(RuntimeError):
-    pass
+    def __init__(self, message, stackname):
+        RuntimeError.__init__(self, message)
```
Comment (lsh-0) [inline src/buildercore/checks.py]: ah, another python3-ism: https://www.journaldev.com/15911/python-super

File: src/tests/test_buildercore_s3.py [hunk 2]
```diff
@@ -3,56 +3,60 @@
...
+class TestReadWrite(base.BaseCase):
     def setUp(self):
-        s3.delete_contents("test/")
-        os.system("mkdir -p /tmp/builder/")
         self.envname = base.generate_environment_name()
 
+        self.prefix = "test/%s" % self.envname
+        s3.delete_contents("%s/" % self.prefix)
+
+        self.local_path = "/tmp/builder-%s/" % self.envname
```
Comment (lsh-0) [inline src/tests/test_buildercore_s3.py]: we should replace this with `buildercore.utils.tempdir`

**Comment by lsh-0** [comment]
replace that tempdir wrangling and I'm happy for this to be merged

File: src/buildercore/checks.py (hunk 1 above)
Comment (giorgiosironi) [reply src/buildercore/checks.py]: can `super` be used in both Python versions?

Comment (lsh-0) [reply src/buildercore/checks.py]: looks like it. Python3:

```python3
Python 3.7.0 (default, Sep 15 2018, 19:13:07) 
[GCC 8.2.1 20180831] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> class Foo:
...     pass
... 
>>> class Bar(Foo):
...     def __init__(self):
...         super(Bar, self).__init__()
... 
>>> baz = Bar()
```

python2:
```python
Python 2.7.15 (default, Jun 27 2018, 13:05:28) 
[GCC 8.1.1 20180531] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> class Foo(object):
...     pass
... 
>>> class Bar(Foo):
...     def __init__(self):
...         super(Bar, self).__init__()
... 
>>> baz = Bar()
```

Comment (lsh-0) [reply src/buildercore/checks.py]: with python2 you have to use 'new style' classes though, like the above

**Pull request merged**